Display message in demo app if no Api Key is set

carlospuk opened the pull request:
Within the demo app, if no ApiKey has been set in StopsModel.swift, display a UIAlert to this effect and inhibit user interaction.

carlospuk committed:
Display message in demo app if no Api Key is set

Within the demo app, if no ApiKey has been set in StopsModel.swift, display a UIAlert to this effect and inhibit user interaction.

carlospuk commented:
![simulator screen shot 19 may 2016 09 49 43](https://cloud.githubusercontent.com/assets/687192/15387843/07eebdb8-1da7-11e6-84d5-5c5e114278d9.png)

urbanthings commented:
Suggest that knowledge of a valid key is kept within the `StopsModel` class. Add a computed var property `hasValidApiKey` or similar for the model to provide details on the key. Test that from the controller. Also should defer the presentation of the alert until main controller presented. Either move alert code to `viewDidAppear` or use `dispatch_after` to delay a small amount to push to a later run loop cycle.

carlospuk committed:
Re-architect solution for alerting on invalid API key

Move detection of invalid API key into StopsModel - note that detection is deliberately indirect to guard against users erroneously replacing the string in the detection func with their API key too!
Present UIAlertController (in the event of invalid API key) in viewDidAppear to ensure container controller is attached

urbanthings commented:
Looks good - thanks
